### ISSUE TITLE
Safetynet Failing

### ISSUE THREAD

**Issue by bggold** [ElderDrivers/EdXposed#133]
Running Riru v16 & EdXposed V0.3.0.0-beta1 breaks safetynet on builds of magisk after 2-28.

Actually just went back to Riru v15 & EdXposed V0.2.9.9-beta2 and still fails.

At this point EdXposed no longer passes safetynet.

**Comment by bggold** [comment]
I can't take credit for this.  Another user posted on XDA that they added google play services to the blacklist in edxposed and that did the trick so far.  Maybe it should be noted in the notes so others know how to resolve the issue.

**Comment by rockyprabowo** [comment]
I'm using 0.3.0.0-beta2, on AOSP 9 based custom ROM with the latest March 2019 security updates and I have no issues with SafetyNet.
Maybe you need to backtrack what are you doing with your device before it breaks.

**Comment by bggold** [comment]
I did and without the modules active it passes. What version of magisk are you running?  You didn't mention.

**Comment by XspeedPL** [comment]
I am using Magisk 18.2 (18112), EdXposed 0.3.0.0-beta2, Riru v16 on Android 8.0

Turned on Magisk Hide for `Google Pay` and `Google Play Services`.
Turned on EdXp blacklist and blacklisted `Google Pay` and `Google Play Services`.

I am using Google Pay without issues.

**Comment by rockyprabowo** [comment]
I'm using the latest stable of Magisk (18.1 build 18100) with Magisk Manager hidden (see screenshot I've linked here )

**Comment by bggold** [comment]
I tried that yesterday and it worked for a little while then failed.  No other changes to phone during that time. I'm running Pie.  That might be part of it.  Right now there are a lot of changes going on in these daily builds with safetynet.  Todays build does not pass safetynet period and yesterdays did.  That is without edxposed enabled.  I'm thinking of just going back to stable v18 on my daily driver for the time being and continue to test on my spare.  Both the same device.

**Comment by bggold** [comment]
rockyprabowo.  There have been no reported issue with v18.1 it seems to be the past week's builds with drastic changes to the way safetynet and magisk hide are being worked by topjohnwu.  Looks like I will go back for now.

**Comment by rockyprabowo** [comment]
Hmm, I remember something that might be related. I was having some issues with Magisk 18.0 weeks ago where Magisk Hide does not doing its job properly by exposing the root access to the app or causing that app to hang indefinitely. It's pretty severe back then to the point that I can't use any app that was hidden from Magisk, so I have to downgrade to v17 Magisk. Looks like it's "partially fixed" on 18.1, I can run them again now, but, after I've tested it again and I found that Magisk Hide will fail silently (no error logged at all) when I have two or more apps that is currently hidden from Magisk running at the same time. As far as I know, SafetyNet pass "fix" depends on Magisk Hide, so if it fails, SafetyNet would fail too. So, I think this problem is Magisk related, not EdXposed/Riru.

**Comment by XspeedPL** [comment]
From a technical standpoint if you're using EdXposed with blacklist the target app isn't being touched.
Therefore it must be an issue with your Magisk Hide.

**Comment by bggold** [comment]
It is an issue with Magisk and EdXposed, more magisk.  There is something that EdXposed is doing that magisk hide does not like.  This is only happening with newer builds.  Yes there are changes going on with magisk hide currently that are breaking things.  I pass all test and can use google pay and other apps that dont work without being listed in magisk hide with no issues without the modules enabled.

**Comment by KaMyKaSii** [comment]
I'm having the same problem on my Zenfone 4 with Android 8.0 and Riru 15 mod + EdXposed 0.3.0.0 beta 1. I can pass safetynet perfectly with all my other Magisk modules enabled (including the Riru 15 mod) except for EdXposed. I have not had a chance to test Riru 16 + EdXposed 0.3.0.0 beta 2 to see if anything changes. I am also using the latest version of Magisk Manager debug + Magisk Canary release, so really a good part of the problem may be on the Magisk side. But for all I know, the Blacklist function simply prohibits EdXposed from touching the ART of the marked app (after all you do not need to de-optimize an app if you do not want to use any modules on it) instead of being a Xposed "Magisk Hide" (as several users seem to think). If Google Play Services can somehow find out that your own ART has been modified by EdXposed, should not this be considered a failure of EdXposed and get more attention?
Anyway I will test the latest version and report the result

**Comment by bggold** [comment]
@KaMyKaSii 

Yes I did the work around again and it seems to be working now.  Not sure why it was failing after the work around yesterday.

I did try beta2 and for some reason it is disabling wifi on both my pixel 3 xl.   I have gone back to beta1 and all is good.

You are correct it has to be something to do with the new commits for magisk hide and the blacklist is a viable work around.  It looks like the issue started after 2-28-19 if that helps.

Thanks
